Add procedure to disable services on OSP side (#407)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_disabling-openstack-services-used-with-stf.adoc

[id="disabling-openstack-services-used-with-stf_{context}"]
= Disabling {OpenStack} services used with {Project}

[role="_abstract"]
Disable the services used when deploying {OpenStack} ({OpenStackShort}) and connecting it to {Project} ({ProjectShort}). Only the services will be disabled. No environment clean up such as logs or generated configuration files will be removed.

.Procedure

. Log in to the {OpenStack} ({OpenStackShort}) undercloud as the `stack` user.

. Source the authentication file:
+
[source,bash]
----
[stack@undercloud-0 ~]$ source stackrc

(undercloud) [stack@undercloud-0 ~]$
----

. Create the `disable-stf.yaml` environment file:
+
[source,yaml,options="nowrap"]
----
(undercloud) [stack@undercloud-0]$ cat > $HOME/disable-stf.yaml <<EOF
---
resource_registry:
  OS::TripleO::Services::CeilometerAgentCentral: OS::Heat::None
  OS::TripleO::Services::CeilometerAgentNotification: OS::Heat::None
  OS::TripleO::Services::CeilometerAgentIpmi: OS::Heat::None
  OS::TripleO::Services::ComputeCeilometerAgent: OS::Heat::None

>   OS::TripleO::Services::Redis: OS::Heat::None

Reviewer: If the environment has Gnocchi running, redis shouldn't be disabled right ? Also while deploying OSP components for STF, it is no where specified to start Redis. It would be good to not change it's state.

Reviewer 2: good catch Yadnesh

Reviewer 2: At the same time, if you have gnocchi running, you also don't want to change the state for Ceilometer

Author: I have accounted for this. That's why I comment at https://github.com/infrawatch/documentation/pull/407/files#diff-d06e423670750ae4c08bf187fa08fd82b1057f96613f7f9131fee6d5f6b9d7b6R44 to place this early in the deployment so that other environment files can override this configuration.

See my comments at https://bugzilla.redhat.com/show_bug.cgi?id=2096853#c3 where I test this with autoscaling and continue to keep the required services running by placing the disable-stf.yaml before the re-enablement.

Reviewer: Yes, those service would get enabled again. However, if one doesn't wish to have all autoscaling components, just ceilometer & gnocchi, that doc wouldn't be referred and Redis wouldn't get enabled. 

Also, we did not say anything about Gnocchi needing Redis in `Sending metrics to Gnocchi` in STF docs. May be we can just leave a comment in `disable-stf.yaml`  to not disable it if using Gnocchi.

Author: I wasn't really going to worry about that part of the documentation, but I think that's a valid point. That section probably needs to be updated or possibly even removed in favour of the new autoscaling documentation.

I'll look and add a note about the gnocchi bits. The disable-stf.yaml is perfectly safe to use with the autoscaling bits as long as you put it as the first thing and then override everything.

Also there is an implicit assumption during documentation writing (for better or worse) that the administrator knows how to use TripleO, meaning we don't need to call out every nuance or document everything multiple times showing how to use the tooling. We just need to provide enough information to help the administrator make decisions.

Point taken though, and I'll look at other sections as well and provide some small note or procedure update.

Author: To wrap this up, I have added a warning to not use this disable procedure with gnocchi. I reviewed again and it's clear what you noted, and I agree.

In the future I want to simply remove the Sending metrics to Gnocchi and Service Telemetry Framework procedure as there should be really no need to run both unless you're doing an autoscaling deployment, in which case you'd just layer the dependencies and environment files per standard deployment procedures. Referencing Gnocchi deployments in STF should not be necessary.

Reviewer: Moving all procedures for legacy telemetry services into autoscaling doc would be really better going forward. From what I see, Tripleo procedure for Gnocchi deployment is documented in STF docs and Aodh is no where to be found. This should consolidate procedures for all legacy telemetry service into one and it also sort of highlights which services we support.

[id="disabling-openstack-services-used-with-stf_{context}"]
= Disabling {OpenStack} services used with {Project}

[role="_abstract"]
Disable the services used when deploying {OpenStack} ({OpenStackShort}) and connecting it to {Project} ({ProjectShort}). Only the services will be disabled. No environment clean up such as logs or generated configuration files will be removed.

[WARNING]

> Do not use this procedure when also using the xref:sending-metrics-to-gnocchi-and-to-stf_assembly-completing-the-stf-configuration[] procedure as the provided `gnocchi-connectors.yaml` does not contain all dependencies required. If removal of {ProjectShort}-related services is desired on {OpenStackShort} then be sure to update your environment to enable data collection and data storage dependencies.

Reviewer 3: ```suggestion
Do not use this procedure when also using the xref:sending-metrics-to-gnocchi-and-to-stf_assembly-completing-the-stf-configuration[] procedure because the `gnocchi-connectors.yaml` does not contain all dependencies required. If you want to remove {ProjectShort}-related services on {OpenStackShort}, ensure that you update your environment to enable data collection and data storage dependencies.
```

Reviewer 3: This whole warning is a little awkward but we can revisit it at a later stage.

.Procedure

> . Log in to the {OpenStack} ({OpenStackShort}) undercloud as the `stack` user.

Reviewer 3: ```suggestion
. Log in to the {OpenStackShort} undercloud as the `stack` user.
```

. Source the authentication file:
+
[source,bash]
----
[stack@undercloud-0 ~]$ source stackrc

(undercloud) [stack@undercloud-0 ~]$
----

. Create the `disable-stf.yaml` environment file:
+
[source,yaml,options="nowrap"]
----
(undercloud) [stack@undercloud-0]$ cat > $HOME/disable-stf.yaml <<EOF
---
resource_registry:
  OS::TripleO::Services::CeilometerAgentCentral: OS::Heat::None
  OS::TripleO::Services::CeilometerAgentNotification: OS::Heat::None
  OS::TripleO::Services::CeilometerAgentIpmi: OS::Heat::None
  OS::TripleO::Services::ComputeCeilometerAgent: OS::Heat::None
  OS::TripleO::Services::Redis: OS::Heat::None
  OS::TripleO::Services::Collectd: OS::Heat::None
  OS::TripleO::Services::MetricsQdr: OS::Heat::None
EOF
----

. Remove the following files from your {OpenStackShort} {OpenStackInstaller} deployment:
+
* `ceilometer-write-qdr.yaml`
* `qdr-edge-only.yaml`
* `enable-stf.yaml`
* `stf-connectors.yaml`

> . Update the {OpenStackShort} overcloud using `openstack overcloud deploy` with the `disable-stf.yaml` early in the list of environment files. If the disable-stf.yaml file is used late in the list of environment files, services that are dependended on by other services could be disabled when not intended. By adding disable-stf.yaml early in the list, other environment files can override the configuration that would disable the service:

Reviewer 3: ```suggestion
. Update the {OpenStackShort} overcloud. Ensure that you use the `disable-stf.yaml` file early in the list of environment files.  By adding `disable-stf.yaml` early in the list, other environment files can override the configuration that would disable the service:
```

Is the second sentence correct? Should it be "cannot override"?

+
// this one is actually a valid use of subs +quotes. We want the underbars to result in emphasis when generated.
+
[source,bash,options="nowrap",subs="+quotes"]
----
(undercloud) [stack@undercloud-0 ~]$ openstack overcloud deploy _<other_arguments>_
--templates /usr/share/openstack-tripleo-heat-templates \
  --environment-file /home/stack/disable-stf.yaml

>   --environment-file _<...other_environment_files...>_ \

Reviewer 3: ```suggestion
  --environment-file _<other_environment_files>_ \
```
